samples/bluetooth/central_iso/src/main.c

```diff
@@ -19,64 +19,75 @@
 #include <zephyr/settings/settings.h>
 #include <zephyr/sys/byteorder.h>
 
+#define LATENCY_MS               10U                   /* 10.0 ms */
+#define INTERVAL_US              (10U * USEC_PER_MSEC) /* 10.0 ms */
+
 static void start_scan(void);
 
 static struct bt_conn *default_conn;
-static struct k_work_delayable iso_send_work;
 static struct bt_iso_chan iso_chan;
-static uint16_t seq_num;
-static uint16_t latency_ms = 10U; /* 10ms */
-static uint32_t interval_us = 10U * USEC_PER_MSEC; /* 10 ms */
 NET_BUF_POOL_FIXED_DEFINE(tx_pool, 1, BT_ISO_SDU_BUF_SIZE(CONFIG_BT_ISO_TX_MTU),
 			  CONFIG_BT_CONN_TX_USER_DATA_SIZE, NULL);
 
 /**
- * @brief Send ISO data on timeout
- *
- * This will send an increasing amount of ISO data, starting from 1 octet.
- *
- * First iteration : 0x00
- * Second iteration: 0x00 0x01
- * Third iteration : 0x00 0x01 0x02
- *
- * And so on, until it wraps around the configured ISO TX MTU (CONFIG_BT_ISO_TX_MTU)
+ * @brief Send ISO data SDU
  *
- * @param work Pointer to the work structure
+ * This will send a decreasing amount of ISO data until it reaches 1 byte then
+ * it wraps around.
  */
-static void iso_timer_timeout(struct k_work *work)
+static void iso_send_sdu(uint32_t seq_num, int32_t next_sdu_ts)
```

Comment:
```suggestion
static void iso_send_sdu(uint32_t seq_num, uint32_t next_sdu_ts, bool is_first_sdu)
```

I think this will make more sense. There's not really any reason to encode the seq_num and "is_first" into the same variable, and since `-1` was only called one place this seems like a better solution to me. 

Furthermore, the timestamp is a `uint32_t`, so for this to have worked the variable should otherwise have been a `int64_t`.

Reply:
This approach is now obsolete, so should be resolved. I returned to a timer-based approach as recommended.